### ISSUE TITLE
Запуск скрипта.

### ISSUE THREAD

**Issue by JohnTriVolta** [askkostya/EgaisMonitoring#4]
Приветствую. Вопрос такой, кто должен выполнять данный скрипт, сам УТМ или же Zabbix-сервер с указанием адреса УТМа в качестве второго параметра скрипта? Допустим УТМов в наличии десятки или даже сотни и устанавливать на каждый из них python и раскидывать скрипты не совсем удобно.

**Comment by k008** [comment]
Здравствуйте, я не автор, но как раз сейчас тоже начал использовать этот скрипт. У автора есть несколько ошибок.
1. Зачем-то указан АтолККТ
2. Поскольку у нас много торговых точек, то более правильным было бы настроить на сервере, а не на каждом клиенте (хотя тогда может ресурса сервера не хватить, но мало вероятно)

Надо удалить узлы Атола, заменить активный агент на внешний скрипт. Ну и произвести настройку соответствующую этим изменениям.

Если исходить из замысла автора, то сервер посылает запрос zabbix агенту, агент выполняет питоновский скрипт и ответ отправляет на сервер. (надо на агентах устанавливать питон. Или как я только на сервере).

Позднее опишу как.

**Comment by JohnTriVolta** [comment]
> Позднее опишу как.

Как сделали? На сервере zabbix поставили python и в конфиге создали под каждый УТМ userparameter?

**Comment by k008** [comment]
> в конфиге создали под каждый УТМ userparameter?

нет, как раз userparameter не надо указывать. Попробуйте по моей инструкции сделать и по изменённым файлам: https://github.com/k008/EgaisMonitoring

**Comment by askkostya** [comment]
В принципе данный скрипт может выполяться откуда угодно.
Первым параметром передаем какие данные хотим считать, вторым путь к УТМ.

У меня как правило сам УТМ расположен непосредственно на кассе или на одной
из касс если в магазине несколько касс, там же расположен и zabbix агент.

Действительно в шаблон zabbix попали метрики от KKTAtol, они у меня в одном шаблоне.
Мониторинг ККТ лежит тут https://github.com/askkostya/KKTATOL

Да, у нас на каждой машине стоит zabbix агент и скрипт выполняет именно он
Инициировать и запускать проверку с сервера достаточно проблематично наверное было бы.
т.к. все проверяемые клиенты за NATом и понадобилось бы опубликовать каждый УТМ.

Python обязателен для работы скрипта, но можно не ставить.
Вот тут лежит portable сборка со всеми необходимыми библиотеками для всех скриптов
https://github.com/askkostya/ZabbixUpdater

Достаточно все содержимое положить в папку C:\zabbix, и в файле .conf прописать путь к своему серверу,
в PATH прописать путь до питона C:\zabbix\files\Python и все скрипты будут работать.
